unstable/wlr-data-control-unstable-v1.xml

```diff
@@ -112,12 +116,17 @@
         The selection event is sent out to notify the client of a new
         wlr_data_control_offer for the selection for this device. The
         wlr_data_control_device.data_offer and the wlr_data_control_offer.offer
-        events are sent out immediately before this event to introduce the data
-        offer object. The selection event is sent to a client when a new
-        selection is set. The wlr_data_control_offer is valid until a new
+        events are sent out before this event to introduce the data offer
+        object. The selection event is sent to a client when a new selection is
+        set.
+
+        For objects of version 1, the wlr_data_control_offer is valid until a new
         wlr_data_control_offer or NULL is received. The client must destroy the
-        previous selection wlr_data_control_offer, if any, upon receiving this
+        previous primary selection wlr_data_control_offer, if any, upon receiving this
         event.
+
+        Starting from version 2, this event does not invalidate the previously
+        received wlr_data_control_offer objects.
```

Comment:
Can't the client rely on the `selection` and `primary_selection` events instead? This would be more consistent with `wl_data_device` (which has selection offers vs. drag-and-drop offers).

Reply:
What if both the primary and the regular clipboard contain the same data offer?

Reply:
I think we should re-use the core protocol wording:

>The data_offer event introduces a new wl_data_offer object, which will subsequently be used in either the data_device.enter event (for drag-and-drop) or the data_device.selection event (for selections).

Reply:
So if a client creates a data offer and then sends it to both `selection` and `primary_selection`, other clients will receive two separate data offers? That would allow keeping the previous invalidate-previous-upon-selection-change semantics.

Reply:
Clients don't create offers. Clients create sources, then calls `set_selection` or `set_primary_selection`, and the compositor will send offers to focused clients via the `selection` and `primary_selection` events.

Reply:
>other clients will receive two separate data offers?

Yes, that is the case.

Reply:
> Clients don't create offers. Clients create sources

Ah, right, forgot about that.